airflow/providers/docker/decorators/docker.py

```diff
@@ -144,7 +144,7 @@ def docker_task(
     return task_decorator_factory(
         python_callable=python_callable,
         multiple_outputs=multiple_outputs,
-        decorated_operator_class=_DockerDecoratedOperator,
+        decorated_operator_class=_DockerDecoratedOperator,  # type: ignore
```

Comment:
This one I think shoiuld be fixable and is definitely worth it - the task_decorator factory might and will be used by others including custom decorators writers, so it's worth to get it right.

Is this because of the multiple inheritance ? @uranusjr @ashb  WDYT?

Reply:
Multiple inheritance _shouldn’t_ be a problem (but hard to tell if it’s a Mypy bug). I’ll try to investigate a bit.

Reply:
Turns out `task_decorator_factory` is just declared wrong. It should annotate `decorated_operator_class` as a `Type[BaseOperator]` instead of `BaseOperator`.

Reply:
Indeed :)

Reply:
Added, thanks.